crates/wasmparser/src/binary_reader.rs

```diff
@@ -316,7 +318,7 @@ impl<'a> BinaryReader<'a> {
 
     pub(crate) fn read_table_type(&mut self) -> Result<TableType> {
         let element_type = self.read_type()?;
-        let flags = self.read_var_u32()?;
+        let flags = self.read_u8()?;
```

Comment:
I think it originally actually did have a factored out method to read limits but memory types have since diverged with the introduction of shared and 64-bit memories, making the limits just table-specific, and I might have inlined the definition here at some point. I think that while limits aren't stored as a separate "thing" this probably makes the most sense, but I refactored the bits and pieces here to avoid the usage of the term `flags` since it's just a byte indicator of whether there's a maximum or not. I'm not actually sure where the `read_var_u32` came from...